library/std/src/sys/unix/os.rs

```diff
@@ -129,12 +129,12 @@ pub fn error_string(errno: i32) -> String {
     }
 }
 
-#[cfg(target_os = "espidf")]
+#[cfg(any(target_os = "espidf", target_os = "horizon"))]
 pub fn getcwd() -> io::Result<PathBuf> {
     Ok(PathBuf::from("/"))
```

Comment:
I was able to run an example that successfully reads the current dir and changes the directory:
```rust
println!("Current dir: {:?}", std::env::current_dir());
println!("Change dir result: {:?}", std::env::set_current_dir("/3ds"));
println!("Current dir: {:?}", std::env::current_dir());
```

I don't think we should remove this.